src/windows_aligned_file_reader.cpp

```diff
@@ -43,9 +45,17 @@ void WindowsAlignedFileReader::register_thread()
         FILE_ATTRIBUTE_READONLY | FILE_FLAG_NO_BUFFERING | FILE_FLAG_OVERLAPPED | FILE_FLAG_RANDOM_ACCESS, NULL);
     if (ctx.fhandle == INVALID_HANDLE_VALUE)
     {
-        std::stringstream ss;
-        ss << m_filename.c_str();
-        diskann::cout << "Error opening " << ss.str() << " -- error=" << GetLastError() << std::endl;
+        const size_t c_max_filepath_len = 256;
```

Comment:
Should this be defined in `defaults.h` or some other location instead of being hardcoded here? Is there any chance of this value being reused in the codebase?

Reply:
Why do we need an upper limit here?

Reply:
wcstombs_s() need pre-allocate char array as input parameter, so we have to allocate a buffer here, we can use vector<char>, but we still need resize() before using it.

Reply:
this function is convert wchar to char, and it only use for print error message, so maybe we can just same the original string as the member of WindowsAlignedFileReader (keep m_filename and m_original_filename).